lib/netext/httpext/tracer.go

```diff
@@ -61,17 +63,17 @@ type Trail struct {
 // SaveSamples populates the Trail's sample slice so they're accesible via GetSamples()
 func (tr *Trail) SaveSamples(tags *stats.SampleTags) {
 	tr.Tags = tags
-	tr.Samples = []stats.Sample{
+	tr.Samples = make([]stats.Sample, 0, 9) // this is with 1 more for a possible HTTPReqFailed
+	tr.Samples = append(tr.Samples, []stats.Sample{
```

Comment:
I don't remember the benchmarks you posted in slack, but does making this a fixed-length array instead of a slice help? 
```suggestion
	tr.Samples = append(tr.Samples, [...]stats.Sample{
```

Reply:
It gives 
```
./tracer.go:67:21: cannot use [...]stats.Sample{...} (type [8]stats.Sample) as type []stats.Sample in append
```
if I add `[:]`
```
./tracer.go:76:3: invalid operation [8]stats.Sample{...}[:] (slice of unaddressable value)
```
I doubt that will make any difference either way ... as I am pretty sure there are enough optimizations specifically for this case ... so it likely doesn't even allocate a slice, to begin with.

The memprofile shows that ... yes SaveSamples is the main (40%) memory used generator, and the only thing that allocates on the heap is the `make`.

As I said in slack if I add 1 more argument to say the slice needs to be bigger the difference is
```
name                                         old time/op    new time/op    delta
MeasureAndEmitMetrics/no_responseCallback-8    6.82µs ± 2%    6.70µs ± 3%    ~     (p=0.310 n=5+5)
MeasureAndEmitMetrics/responseCallback-8       7.16µs ± 3%    6.97µs ± 3%    ~     (p=0.095 n=5+5)
name                                         old alloc/op   new alloc/op   delta
MeasureAndEmitMetrics/no_responseCallback-8    1.10kB ± 0%    1.03kB ± 0%  -5.84%  (p=0.008 n=5+5)
MeasureAndEmitMetrics/responseCallback-8       1.10kB ± 0%    1.10kB ± 0%    ~     (all equal)
name                                         old allocs/op  new allocs/op  delta
MeasureAndEmitMetrics/no_responseCallback-8      9.00 ± 0%      9.00 ± 0%    ~     (all equal)
MeasureAndEmitMetrics/responseCallback-8         9.00 ± 0%      9.00 ± 0%    ~     (all equal)
```
regardless of whether with a single boolean or with an integer to be added to the length. I can add this as it does seem helpful ... but not by much